test/values.test.js

```diff
@@ -501,12 +538,228 @@ describe('Values', () => {
+    assertPrint(
+      new Query({ lambda: 'X', expr: { var: 'X' }, api_version: '3' }),
+      'Query(Lambda("X", Var("X")))'
+    )
+  })
+
+  test('pretty print despite argument order', () => {
+    // Match
+    assertPrint(
+      new Query({
+        api_version: '3',
+        expr: {
+          match: {
+            index: 'idx',
+          },
+          terms: ['str', 10],
+        },
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), ["str", 10])))'
+    )
+
+    assertPrint(
+      new Query({
+        expr: {
+          terms: 10,
+          match: {
+            index: 'idx',
+          },
+        },
+        api_version: '3',
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), 10)))'
+    )
+
+    // Filter an array
+    assertPrint(
+      new Query({
+        collection: [1, 2, 3],
+        filter: {
+          expr: {
+            equals: [0, { modulo: [{ var: 'i' }, 2] }],
+          },
+          lambda: 'i',
+        },
+      }),
+      'Query(Filter([1, 2, 3], Lambda("i", Equals(0, Modulo(Var("i"), 2)))))'
+    )
+
+    // Filter a page
+    assertPrint(
+      new Query({
+        filter: {
+          lambda: 'i',
+          expr: {
+            containsstr: { select: 'name', from: { get: { var: 'i' } } },
+            search: '-',
+          },
+        },
+        collection: { paginate: { databases: null } },
+      }),
+      'Query(Filter(Paginate(Databases()), Lambda("i", ContainsStr(Select("name", Get(Var("i"))), "-"))))'
+    )
+
+    // Select
+    assertPrint(
+      new Query({
+        from: {
+          object: {
+            favorites: {
+              object: {
+                foods: ['crunchings', 'munchings', 'lunchings'],
+              },
+            },
+          },
+        },
+        select: ['favorites', 'foods', 1],
+      }),
+      'Query(Select(["favorites", "foods", 1], {favorites: {foods: ["crunchings", "munchings", "lunchings"]}}))'
+    )
+
+    // Map an Array
+    assertPrint(
+      new Query({
+        map: { lambda: 'x', expr: { add: [{ var: 'x' }, 1] } },
+        collection: [1, 2, 3],
+      }),
+      'Query(Map([1, 2, 3], Lambda("x", Add(Var("x"), 1))))'
+    )
+
+    // Map a Page
+    new assertPrint(
+      new Query({
+        map: {
+          expr: { get: { var: 'x' } },
+          lambda: 'x',
+        },
+        collection: { paginate: { databases: null } },
+      }),
+      'Query(Map(Paginate(Databases()), Lambda("x", Get(Var("x")))))'
+    )
+
+    // Foreach
+    assertPrint(
+      new Query({
+        collection: { paginate: { documents: { collection: 'orders' } } },
+        foreach: {
+          lambda: 'doc',
+          expr: {
+            update: { var: 'doc' },
+            params: { object: { data: { object: { createdBy: 'Ryan' } } } },
+          },
+        },
+      }),
+      `Query(Foreach(Paginate(Documents(Collection("orders"))), Lambda("doc", Update(Var("doc"), {data: {createdBy: "Ryan"}}))))`
```

Comment:
@marrony I hit some of your notes re: Databases, Collections and Documents, however, this test is now failing. For some reason, the Documents() never displays the Collection("orders") inside it.

I've tracked the issue to the `if ('paginate' in expr)` block of Expr.js, but am not sure why it is disregarding my current logic (which I think should work) 🤔